ci: add documentation check to documentation workflow

mhofstetter opened the pull request:
Currently, differences in generated documentation (codeowners, cmd-ref, helm, ...) are only checked somewhat hidden in the Travis workflow; and only due to a transitive dependency when executing `make` in `.travis/build.sh`.

This might not be the place where a PR author and reviewer is expecting this to be checked.

Therefore, this PR adds the documentation checks as separate job to the documentation workflow.

mhofstetter committed:
ci: add documentation check to documentation workflow

Currently, differences in generated documentation are only checked
in the Travis workflow; and only due to a transitive dependency when
executing `make` in `.travis/build.sh`.

Therefore, this commit executes the checks in the documentation workflow.

Signed-off-by: Marco Hofstetter <marco.hofstetter@isovalent.com>

mhofstetter committed:
build: mark codeowners.rst as phony target

This commit marks the make target `codedowners.rst` (Documentation)
as phony target. This way, the generated gets triggered independent of
changes in the CODEOWNERS file.

Signed-off-by: Marco Hofstetter <marco.hofstetter@isovalent.com>

brlbil commented on .github/workflows/documentation.yaml:
```diff
@@ -65,3 +65,49 @@ jobs:
           args: html
+
+  check-generated-documentation:
+    name: Check generated documentation
```
This workflow has a trigger for merge_queue, so this job will be triggered when a PR is added to the queue.
This will not break merge_queue, but it is wasteful unless this job is added as a required check.

@aanm Do you want to add this check to the required checks? We do not have the historical run data, but the last run took 5m40s. It seems ok.

If we do not want to add the job to the required checks then I suggest adding the if statement below to skip it in merge_queue runs.

```
 if: ${{ github.event_name != 'merge_group' }}
```

aanm replied:
Travis is also not checked on merge_group thus I would suggest to not add this one as well.

mhofstetter replied:
Thanks for the hint @brlbil .

I added the condition.

aanm reviewed (commented):
@mhofstetter if we are going to have this dedicated check perhaps we should remove the one on travis.

mhofstetter committed:
ci: don't execute postcheck on integration tests (travis & gha)

Currently, part of the integration test execution is executing
`make` in `.travis/build.sh`. The default target (`all`) implicitly
executes `precheck`, `build` & `postcheck`.

With the execution of the postcheck in the documentation workflow,
it's not longer necessary to execute them in the integrations too.

Therefore, this commit removes its execution.

Signed-off-by: Marco Hofstetter <marco.hofstetter@isovalent.com>

learnitall reviewed (approved):
This is awesome, thank you!
